### ISSUE TITLE
Sweep_all command not finding all inputs on v7

### ISSUE THREAD

**Issue by xmr-eric** [monero-project/monero#2957]
I typed `sweep_all` on the v7 testnet into my CLI with a balance of 1018 XMR. The sweep_all command only found a single input for 0.01 XMR. I typed `refresh` and my balance was still 1018 XMR. Then I tried `transfer address 1000` and that worked fine. Is sweep_all not finding all the inputs on v7 testnet?

https://paste.fedoraproject.org/paste/EonuDL7TAasOUPui17gpJw

**Comment by stoffu** [comment]
Do you have funds received in subaddresses? You can see it by `balance detail`. If this is the case, `sweep_all` command spends funds associated with just a single subaddress unless you explicitly specify spending subaddresses' indices like `index=1,2,3`.

**Comment by xmr-eric** [comment]
Yep that’s the issue.

**Comment by farinspace** [comment]
I don't quite understand "not finding all the inputs" ... but have experienced this problem twice already.

I restored a wallet with only a single inbound transaction. I then tried to move all XMR using `sweep_all [addr]`, at this point the wallet freezes. I have to `kill -9 [pid]` to stop it.

I then do `transfer [addr] [small-amount]` and it works, then I do `sweep_all [remaining-amount]` and then it works.

Not sure if this is related, but wouldn't mind understanding why this happens. thx.
